pkg/build/builder/sti.go

```diff
@@ -188,6 +198,9 @@ func (s *S2IBuilder) Build() error {
 		Injections:                injections,
 		ScriptDownloadProxyConfig: scriptDownloadProxyConfig,
 		BlockOnBuild:              true,
+		RuntimeImage:              runtimeImageName,
+		RuntimeArtifacts:          runtimeArtifacts,
+		RuntimePullSecret:         runtimePullSecret,
```

Comment:
I've added a field `RuntimeAuthentication`. As far I understand you shouldn't pass a secret to S2I directly, instead you're passing auth info. I've found this example and probably you could do the same: https://github.com/openshift/origin/blob/2a2a8b9dfdc437c5b210fd7b748294ff4afe87ff/pkg/build/builder/sti.go#L227-L228